### PR TITLE
cmd/tailscale/cli: rename share command to drive

### PR DESCRIPTION
**Pull request by oxtoacart**
Updates tailscale/corp#16827

```
> ./tailscale drive -h
USAGE
  drive share <name> <path>
  drive rename <oldname> <newname>
  drive unshare <name>
  drive list

Tailscale drive allows you to share directories with other machines on your tailnet.

In order to share folders, your node needs to have the node attribute "drive:share".

In order to access shares, your node needs to have the node attribute "drive:access".

For example, to enable sharing and accessing shares for all member nodes:

  "nodeAttrs": [
    {
      "target": ["autogroup:member"],
      "attr": [
        "drive:share",
        "drive:access",
      ],
    }]

Each share is identified by a name and points to a directory at a specific path. For example, to share the path /Users/me/Documents under the name "docs", you would run:

  $ tailscale drive share docs /Users/me/Documents

Note that the system forces share names to lowercase to avoid problems with clients that don't support case-sensitive filenames.

Share names may only contain the letters a-z, underscore _, parentheses (), or spaces. Leading and trailing spaces are omitted.

All Tailscale shares have a globally unique path consisting of the tailnet, the machine name and the share name. For example, if the above share was created on the machine "mylaptop" on the tailnet "mydomain.com", the share's path would be:

  /mydomain.com/mylaptop/docs

In order to access this share, other machines on the tailnet can connect to the above path on a WebDAV server running at 100.100.100.100:8080, for example:

  http://100.100.100.100:8080/mydomain.com/mylaptop/docs

Permissions to access shares are controlled via ACLs. For example, to give yourself read/write access and give the group "home" read-only access to the above share, use the below ACL grants:

  "grants": [
    {
      "src": ["mylogin@domain.com"],
      "dst": ["mylaptop's ip address"],
      "app": {
        "tailscale.com/cap/drive": [{
          "shares": ["docs"],
          "access": "rw"
        }]
      }
    },
    {
      "src": ["group:home"],
      "dst": ["mylaptop"],
      "app": {
        "tailscale.com/cap/drive": [{
          "shares": ["docs"],
          "access": "ro"
        }]
      }
    }]

To categorically give yourself access to all your shares, you can use the below ACL grant:

  "grants": [
    {
      "src": ["autogroup:member"],
      "dst": ["autogroup:self"],
      "app": {
        "tailscale.com/cap/drive": [{
          "shares": ["*"],
          "access": "rw"
        }]
      }
    }]

Whenever either you or anyone in the group "home" connects to the share, they connect as if they are using your local machine user. They'll be able to read the same files as your user and if they create files, those files will be owned by your user.

If you want a share to be accessed as a different user, you can use sudo to accomplish this. For example, to create the aforementioned share as "theuser", you could run:

	$ sudo -u theuser tailscale drive share docs /Users/theuser/Documents

You can rename shares, for example you could rename the above share by running:

  $ tailscale drive rename docs newdocs

You can remove shares by name, for example you could remove the above share by running:

  $ tailscale drive unshare newdocs

You can get a list of currently published shares by running:

  $ tailscale drive list

SUBCOMMANDS
  share    [ALPHA] create or modify a share
  rename   [ALPHA] rename a share
  unshare  [ALPHA] remove a share
  list     [ALPHA] list current shares
```

### CODE REVIEW

**Review by catzkorn — APPROVED** [review]

File: cmd/tailscale/cli/share.go [hunk 1]
```diff
@@ -57,14 +58,14 @@ var shareCmd = &ffcli.Command{
...
 	},
 	Exec: func(context.Context, []string) error {
-		return errors.New("share subcommand required; run 'tailscale share -h' for details")
+		return errors.New("drive subcommand required; run 'tailscale drive -h' for details")
 	},
 }
 
-// runShareSet is the entry point for the "tailscale share set" command.
+// runShareSet is the entry point for the "tailscale drive share" command.
 func runShareSet(ctx context.Context, args []string) error {
```
Comment (bradfitz) [inline cmd/tailscale/cli/share.go:66]: not renaming the func?

File: cmd/tailscale/cli/share.go [hunk 2]
```diff
@@ -151,7 +152,7 @@ func buildShareLongHelp() string {
 	return fmt.Sprintf(shareLongHelpBase, longHelpAs)
 }
 
-var shareLongHelpBase = `Tailscale share allows you to share directories with other machines on your tailnet.
+var shareLongHelpBase = `Tailscale drive allows you to share directories with other machines on your tailnet.
```
Comment (bradfitz) [inline cmd/tailscale/cli/share.go:155]: Taildrive? Tailscale Drive?

File: cmd/tailscale/cli/share.go [hunk 3]
```diff
@@ -108,10 +109,10 @@ func runShareRename(ctx context.Context, args []string) error {
 	return err
 }
 
-// runShareList is the entry point for the "tailscale share list" command.
+// runShareList is the entry point for the "tailscale drive list" command.
 func runShareList(ctx context.Context, args []string) error {
```
Comment (bradfitz) [inline cmd/tailscale/cli/share.go:113]: kinda weird for "tailscale drive list" to have no "share" in it but this func to have "share" in it (runShareList).

File: cmd/tailscale/cli/share.go [hunk 4]
```diff
@@ -14,27 +14,28 @@ import (
 )
```
Comment (bradfitz) [inline cmd/tailscale/cli/share.go:14]: gonna rename this whole file from share.go to drive.go too?

File: cmd/tailscale/cli/share.go (hunk 2 above)
Comment (oxtoacart) [reply cmd/tailscale/cli/share.go]: Fixed

File: cmd/tailscale/cli/share.go (hunk 3 above)
Comment (oxtoacart) [reply cmd/tailscale/cli/share.go]: Fixed.

**Pull request merged**